packages/core/src/presentation/Presentation.ts

```diff
@@ -194,5 +235,274 @@ export function create(
 
   validateStructure(presentation)
   assertHolderCanPresentCredentials(presentation)
-  return presentation
+
+  if (!signers) {
+    return presentation
+  }
+
+  const holderDocument =
+    typeof holder === 'string' ? (await resolve(holder)).didDocument : holder
+
+  if (!holderDocument?.id) {
+    throw new SDKErrors.DidNotFoundError(
+      `Failed to resolve holder DID ${holderDid}`
+    )
+  }
+
+  const requiredAlgorithms = cryptosuites.map(
+    ({ requiredAlgorithm }) => requiredAlgorithm
+  )
+
+  const signer = await Signers.selectSigner(
+    signers,
+    byAlgorithm(requiredAlgorithms),
+    byDid(holderDocument, {
+      verificationRelationship: purpose,
+      controller: holderDid,
+    })
+  )
+  if (!signer) {
+    throw new SDKErrors.NoSuitableSignerError(undefined, {
+      signerRequirements: {
+        algorithm: requiredAlgorithms,
+        did: holderDid,
+        verificationRelationship: purpose,
+      },
+      availableSigners: signers,
+    })
+  }
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- We've matched the suite to the algorithms earlier, so this will return the right suite
+  const suite = cryptosuites.find(
+    ({ requiredAlgorithm }) => requiredAlgorithm === signer.algorithm
+  )!
+  return DataIntegrity.createProof(presentation, suite, signer, {
+    purpose,
+    challenge,
+    domain: verifier,
+    // TODO: it's unclear what the canonical way of identifying the intended audience for the presentation should be;
+    // The `verifier` claim on the presentation or the `domain` option on the proof.
+    // Duplicating this mechanism may help support different tooling?
```

Comment:
Then why not taking the `domain` as an input parameter to the function?

Reply:
Yeah makes sense to leave it open like take for the time being. Will do

Reply:
24c11745da83fdaca9a7bae229c0b9c7cca51a19